cmd/ilm/options.go

```diff
@@ -40,10 +40,15 @@ func RemoveILMRule(lfcCfg *lifecycle.Configuration, ilmID string) (*lifecycle.Co
 		return lfcCfg, probe.NewError(fmt.Errorf("lifecycle configuration not set"))
 	}
 	n := 0
+	var expiryRuleRemoved bool
 	for _, rule := range lfcCfg.Rules {
 		if rule.ID != ilmID {
 			lfcCfg.Rules[n] = rule
 			n++
+		} else {
+			if !rule.Expiration.IsNull() {
+				expiryRuleRemoved = true
+			}
```

Comment:
can you detect this on the server side? It is brittle to do this on client - an older mc client can be used to update ilm rules and changes won't be propagated correctly.

Reply:
As the said rule is removed from the list and rest are sent for update and also in remove all scenario an empty list of rules sent for update call, I feel flag is the way to tell backend that as part this update, there was removal of expiry rule involved.
This could have been possible if delete was altogether a new API taking list of deleted rule ids. May be we can think of that as new change.

Reply:
you should be able to get current config if it exists prior to the update and compare with incoming lfc config and determine if an expiry rule was removed - `mc` client may/may note be the latest, and we should not rely on client side code for enforcing server functionality.

Reply:
makes sense. let me try this.